Content/Items/Misc/Accessories.TaintedGreataxe.cs

```diff
@@ -0,0 +1,486 @@
+                {
+                    Vector2 velocity = Projectile.velocity.RotatedByRandom(MathHelper.ToRadians(35f)) * Main.rand.NextFloat(-0.9f, -0.15f);
+                    Dust.NewDustDirect(Projectile.position, 4, 4, ModContent.DustType<TaintedGreataxeDust>(), velocity.X, velocity.Y, 40 + Main.rand.Next(40), default, Main.rand.NextFloat(0.7f, 1f)).
+                        rotation = Main.rand.NextFloat(6.18f);
+                }
+
+                CameraSystem.Shake += 11;
+
+                Projectile.rotation = MathHelper.ToRadians(Main.rand.NextFloat(45f, 60f)) * Projectile.direction;
+
+                stickyAI = true;
+                Projectile.friendly = false;
+                enemyWhoAmI = target.whoAmI;
+                offset = Projectile.position - target.position;
+                offset -= Projectile.velocity;
+                Projectile.netUpdate = true;
+            }
+        }
+
+        public override bool? CanHitNPC(NPC target)
+        {
+            return Embedding && target == targetNPC && !stickyAI;
+        }
+
+        public override void SendExtraAI(BinaryWriter writer)
+        {
+            writer.Write(stickyAI);
+            writer.WritePackedVector2(offset);
+            writer.Write(enemyWhoAmI);
+        }
+
+        public override void ReceiveExtraAI(BinaryReader reader)
+        {
+            stickyAI = reader.ReadBoolean();
+            offset = reader.ReadPackedVector2();
+            enemyWhoAmI = reader.ReadInt32();
+        }
+
+        public override void Kill(int timeLeft)
+        {
+            Player.GetModPlayer<TaintedGreataxePlayer>().GreatAxeProjectileWhoAmI = -1;
+        }
+
+        private void EmbeddingAI()
+        {
+            if (!targetNPC.active || Projectile.Distance(Player.Center) > 1600f) //un embed if too far away from player
+            {
+                Embedding = false;
+                stickyAI = false;
+                Projectile.velocity = Vector2.UnitY;
+                EmbeddingCooldown = 120;
+            }
+
+            if (!stickyAI)
+            {
+                Projectile.rotation += Projectile.velocity.X * 0.025f;
+
+                Vector2 direction = targetNPC.Center - Projectile.Center;
+                if (direction.Length() > 25)
+                {
+                    direction.Normalize();
+                    Projectile.velocity = Vector2.Lerp(Projectile.velocity, direction * 35f, 0.02f);
+                }
+
+                Dust.NewDustPerfect(Projectile.Center, ModContent.DustType<Dusts.GlowFastDecelerate>(), null, 0, new Color(85, 220, 55), 0.35f);
+            }
+            else
+            {
+                Dust.NewDust(targetNPC.position, targetNPC.width, targetNPC.height, ModContent.DustType<Dusts.GlowFastDecelerate>(), 0, -2.5f, newColor: new Color(85, 220, 55), Scale: 0.4f);
+
+
+                if (Main.mouseRight && Main.mouseRightRelease)// right clicking the projectile unembeds it
```

Comment:
is there a reason these conditionals are chained like this? lets favor using logical operators over chaining like this